File: FluentFTP.CSharpExamples/UploadFileWithProgress.cs
using System;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using FluentFTP;

namespace Examples {

	internal static class UploadFileWithProgressExample {

		public static void UploadFile() {
			using (var ftp = new FtpClient("127.0.0.1", "ftptest", "ftptest")) {
				ftp.Connect();

				// define the progress tracking callback
				Action<FtpProgress> progress = delegate (FtpProgress p) {
					if (p.Progress == 1) {
						// all done!
					}
					else {
						// percent done = (p.Progress * 100)
					}
				};

				// upload a file with progress tracking
				ftp.UploadFile(@"D:\Github\FluentFTP\README.md", "/public_html/temp/README.md", FtpRemoteExists.Overwrite, false, FtpVerify.None, progress);

			}
		}

		public static async Task UploadFileAsync() {
			var token = new CancellationToken();
			using (var ftp = new AsyncFtpClient("127.0.0.1", "ftptest", "ftptest")) {
				await ftp.Connect(token);

				// define the progress tracking callback
				Progress<FtpProgress> progress = new Progress<FtpProgress>(p => {
					if (p.Progress == 1) {
						// all done!
					}
					else {
						// percent done = (p.Progress * 100)
					}
				});

				// upload a file with progress tracking
				await ftp.UploadFile(@"D:\Github\FluentFTP\README.md", "/public_html/temp/README.md", FtpRemoteExists.Overwrite, false, FtpVerify.None, progress, token);

			}
		}

	}
}

File: FluentFTP.VBExamples/UploadFileWithProgress.vb
Imports System
Imports System.Net
Imports System.Threading
Imports System.Threading.Tasks
Imports FluentFTP

Namespace Examples
	Friend Module UploadFileWithProgressExample
		Sub UploadFile()
			Using ftp = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				ftp.Connect()

				' define the progress tracking callback
				Dim progress As Action(Of FtpProgress) =
					Sub(ByVal p As FtpProgress)
						If p.Progress = 1 Then
							' all done!
						Else
							' percent done = (p.Progress * 100)
						End If
					End Sub

				' upload a file with progress tracking
				ftp.UploadFile("D:\Github\FluentFTP\README.md", "/public_html/temp/README.md", FtpRemoteExists.Overwrite, False, FtpVerify.None, progress)

			End Using
		End Sub

		Async Function UploadFileAsync() As Task
			Dim token = New CancellationToken()

			Using ftp = New FtpClient("127.0.0.1", "ftptest", "ftptest")
				Await ftp.ConnectAsync(token)

				' define the progress tracking callback
				Dim progress As Progress(Of FtpProgress) = New Progress(Of FtpProgress)(
					Sub(p)
						If p.Progress = 1 Then
							' all done!
						Else
							' percent done = (p.Progress * 100)
						End If
					End Sub)

				' upload a file with progress tracking
				Await ftp.UploadFileAsync("D:\Github\FluentFTP\README.md", "/public_html/temp/README.md", FtpRemoteExists.Overwrite, False, FtpVerify.None, progress, token)

			End Using
		End Function
	End Module
End Namespace
